Added an example with finding optimal WC hyperparams (#2334)
openvinotoolkit/nncf: examples/weight_compression/openvino/tiny_llama/README.md

# Weight compression of TinyLLama OpenVINO Model

> This example demonstrates how to find optimal ratio and group size to compress weights of the TinyLLama model from the HuggingFace Transformers framework. OpenVINO backend supports mixed precision weight compression with a 4-bit data type as a primary precision. The fastest mixed-precision mode is `INT4_SYM`, but it may lead to a significant accuracy degradation. In this example, the allowed maximum deviation from the original model is `0.2`. If the similarity of the compressed model is not satisfying, there are 2 hyper-parameters to tune: `group_size` and `ratio`. Lower group size and less ratio of 4-bit layers usually improve accuracy at the sacrifice of inference speed.

Reviewer: ```suggestion
This example demonstrates how to find the appropriate `ratio` and `group_size` parameters to compress the weights of the TinyLLama model from the HuggingFace Transformers. OpenVINO backend supports inference of mixed-precision models with weights compressed to a 4-bit data type as a primary precision. The fastest mixed-precision mode is `INT4_SYM`, but it may lead to a significant accuracy degradation, especially for models of moderate size. In this example, the allowed maximum deviation from the original model is `0.2` points of the similarity metric. If the similarity of the compressed model is not satisfying, there are 2 hyper-parameters to tune: `group_size` and `ratio`. Smaller `group_size` and `ratio` of 4-bit layers usually improve accuracy at the sacrifice of model size and inference latency.
```

The example includes the following steps:

> - Download and prepare `wikitext` dataset.

Reviewer: ```suggestion
- Prepare `wikitext` dataset.
```

> - Download and export `PY007/TinyLlama-1.1B-step-50K-105b` model to OpenVINO IR.

Reviewer: ```suggestion
- Prepare `PY007/TinyLlama-1.1B-step-50K-105b` text-generation model in OpenVINO representation using Optimum-Intel.
```

- Compress weights of the model with NNCF Weight compression algorithm.

> - Find optimal ratio and group size if acceptable similarity is not achieved.

Reviewer: ```suggestion
- Find appropriate `ratio` and `group_size` if acceptable similarity is not achieved.
```

> - Save model with the best hyperparameters found.
> - Measure similarity and footprint of the compressed model.

Reviewer: ```suggestion
- Measure the similarity and footprint of the final model.
```

## Install requirements

> At this point it is assumed that you have already installed NNCF. You can find information on installation NNCF [here](https://github.com/openvinotoolkit/nncf#user-content-installation).

Reviewer: Let's don't do this assumption and consider the user starting from scratch.
